contracts/DefaultStrategy.sol

```diff
@@ -522,248 +436,288 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
+            revert RebalanceNoExtraStCelo(fromGroup, actualFromStCelo, expectedFromStCelo);
+        }
+
+        (uint256 expectedToStCelo, uint256 actualToStCelo) = getExpectedAndActualStCeloForGroup(
+            toGroup
+        );
+
+        if (actualToStCelo >= expectedToStCelo) {
+            // toGroup needs to have less stCELO than it should
+            revert RebalanceEnoughStCelo(toGroup, actualToStCelo, expectedToStCelo);
         }
 
-        return votableGroupsFinal;
+        uint256 toMove = Math.min(
+            actualFromStCelo - expectedFromStCelo,
+            expectedToStCelo - actualToStCelo
+        );
+
+        updateGroupStCelo(fromGroup, toMove, false);
+        updateGroupStCelo(toGroup, toMove, true);
+        if (sorted) {
+            (address lesserKey, address greaterKey) = getLesserAndGreaterOfActiveGroupsWithdrawal(
+                fromGroup,
+                stCELOInGroup[fromGroup]
+            );
+            if (lesserKey != greaterKey) {
+                activeGroups.update(fromGroup, stCELOInGroup[fromGroup], lesserKey, greaterKey);
+                (lesserKey, greaterKey) = getLesserAndGreaterOfActiveGroupsDeposit(
+                    toGroup,
+                    stCELOInGroup[toGroup]
+                );
+                if (lesserKey != greaterKey) {
+                    activeGroups.update(toGroup, stCELOInGroup[toGroup], lesserKey, greaterKey);
+                    return;
+                }
+            }
+            sorted = false;
+        }
+        unsortedGroups.add(toGroup);
+        unsortedGroups.add(fromGroup);
     }
 
     /**
-     * @notice Calculates how many votes should be withdrawn from each
-     * deprecated group.
-     * @param withdrawal The total amount of votes that needs to be withdrawn.
-     * @return deprecatedGroupsWithdrawn The array of deprecated groups to be
-     * withdrawn from.
-     * @return deprecatedWithdrawalsPerGroup The amount of votes to withdraw
-     * from the respective deprecated group in `deprecatedGroupsWithdrawn`.
-     * @return numberDeprecatedGroupsWithdrawn The number of groups in
-     * `deprecatedGroupsWithdrawn` that have a non zero withdrawal.
-     * @return remainingWithdrawal The number of votes that still need to be
-     * withdrawn after withdrawing from deprecated groups.
-     * @dev Non zero entries of `deprecatedWithdrawalsPerGroup` will be exactly
-     * a prefix of length `numberDeprecatedGroupsWithdrawn`.
-     */
-    function getDeprecatedGroupsWithdrawalDistribution(uint256 withdrawal)
-        internal
-        returns (
-            address[] memory deprecatedGroupsWithdrawn,
-            uint256[] memory deprecatedWithdrawalsPerGroup,
-            uint256 numberDeprecatedGroupsWithdrawn,
-            uint256 remainingWithdrawal
-        )
+     * @notice Returns expected stCELO vs actual stCELO for group.
+     * @param group The group.
+     * @return expectedStCelo The stCELO which group should have.
```

Comment:
```suggestion
     * @return expectedStCelo The amount of stCELO which group should have.
```
And maybe add an explanation of what that means exactly, e.g. "(the total amount of stCELO in the default strategy divided by the number of active groups)."